### PR TITLE
Use ifgroups

### PR DESCRIPTION
**Pull request by Savolro**

### CODE REVIEW

File: daemon/routes/ifgroup/ifgroup.go [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+const Group = 0xe1f1
+
+var (
+	// ErrAlreadySet defines that NetlinkManager is already set.
+	ErrAlreadySet = errors.New("already set")
+	// ErrNotSet defines that NetlinkManager is not set yet.
+	ErrNotSet = errors.New("not set")
+)
+
+// Manager is responsible for manager ifgroup to the network interfaces
```
Comment (bartoszWojciechO) [inline daemon/routes/ifgroup/ifgroup.go:22]: Is this comment worded correctly?

File: daemon/routes/iprule/iprule.go [hunk 2]
```diff
@@ -400,131 +339,94 @@ func removeFwmarkRule(fwMarkVal uint32, ipv6 bool) error {
...
-		"lookup",
-		"main",
-		"suppress_prefixlength",
-		"0",
-	}
-
-	// #nosec G204 -- input is properly sanitized
-	if out, err := exec.Command(cmdStr, cmdParams...).CombinedOutput(); err != nil {
-		return fmt.Errorf("executing '%s %s' command: %w: %s", cmdStr, strings.Join(cmdParams, " "), err, string(out))
+// addSuppressRule create/add suppress_prefixlength rule
```
Comment (keliramu) [inline daemon/routes/iprule/iprule.go:348]: It's not `suppress_prefixlength` anymore

File: daemon/routes/ifgroup/ifgroup.go (hunk 1 above)
Comment (bartoszWojciechO) [inline daemon/routes/ifgroup/ifgroup.go:22]: Is this comment correctly worded?

File: daemon/routes/ifgroup/ifgroup.go [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+type NetlinkManager struct {
+	group      int
+	deviceList func() ([]net.Interface, error)
+	set        bool
+	backup     map[int]uint32
+	mu         sync.Mutex
+}
+
+// NewNetlinkManager is a default constructor for NetlinkManager
+func NewNetlinkManager(deviceList func() ([]net.Interface, error)) *NetlinkManager {
```
Comment (bartoszWojciechO) [inline daemon/routes/ifgroup/ifgroup.go:39]: Will there be uts implemented for NetlinkManager later on? If not, maybe `ListPhysical` could be just called directly?

Comment (Savolro) [reply daemon/routes/ifgroup/ifgroup.go:39]: Integration `Link` test is implemented. It still uses `ListPhysical` but any other function could be used

**Review by piotrjurkiewicz — APPROVED** [review]

**Review by keliramu — APPROVED** [review]

**Pull request merged**